sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer.implementation;
+
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * Utility method class.
+ */
+public class Utility {
+    private static final ClientLogger LOGGER = new ClientLogger(Utility.class);
+
+    /**
+     * A utility method for converting the input stream to Flux of ByteBuffer.
+     *
+     * @param data The input data which needs to convert to ByteBuffer.
+     *
+     * @return {@link ByteBuffer} which contains the input data.
+     * @throws RuntimeException When I/O error occurs.
+     */
+    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data) {
+        try {
+            return Flux.just(toByteArray(data));
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    private static ByteBuffer toByteArray(InputStream in) throws IOException {
+
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        byte[] buffer = new byte[1024];
+        int len;
+
+        // read bytes from the input stream and store them in buffer
+        while ((len = in.read(buffer)) != -1) {
+            // write bytes from the buffer into output stream
+            os.write(buffer, 0, len);
+        }
+
+        return ByteBuffer.wrap(os.toByteArray());
+    }
```

Comment:
Do we want to convert the entire `InputStream` to a `ByteBuffer` in a single go or should this be chunked as downstream requests more elements? Could we have scenarios where a large file is being read that wouldn't fit into memory?

Reply:
I have added a separate issue to track this #9690 
And if it helps, the service docs do mention a size limit specification 
>Image file size must be less than 20 MB.

I will also add it to our javadocs.

Reply:
The service will only accept files of 20MB, so anything larger is an exception, but nothing is stopping me from sends a 2GB file by accident.

Reply:
Updating this implementation in #9690